### PR TITLE
feat: Add routeoptimization snippets

### PR DESCRIPTION
**Pull request by bshi**
## Description

Note: **This PR originates from a fork; I don't have permission to edit this repository. This is self-service request - please advise**

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [x] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [X] These samples need a new **API enabled** in testing projects to pass (let us know which ones) **routeoptimization.googleapis.com**
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork. **I don't have permission on this repository - this is self-service request - please advise**
- [X] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [X] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -69,3 +69,4 @@ security-center @GoogleCloudPlatform/gcp-security-command-center @GoogleCloudPla
 retail @GoogleCloudPlatform/cloud-retail-team @GoogleCloudPlatform/nodejs-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
 media @GoogleCloudPlatform/cloud-media-team @GoogleCloudPlatform/nodejs-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
 healthcare @GoogleCloudPlatform/healthcare-life-sciences @GoogleCloudPlatform/nodejs-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
+routeoptimization @GoogleCloudPlatform/geo-routeoptimization @GoogleCloudPlatform/nodejs-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
```
Comment (subfuzion) [inline CODEOWNERS:72]: Hi @bshi Please confirm team exists:
> Unknown owner on line 72: make sure the team @GoogleCloudPlatform/geo-routeoptimization exists, is publicly visible, and has write access to the repository
routeoptimization @GoogleCloudPlatform/geo-routeoptimization @GoogleCloudPlatform/nodejs-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
Original file line numb

File: routeoptimization/snippets/optimizeTours.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  // [END routeoptimization_optimize_tours_basic]
+
+  console.log(JSON.stringify(await callOptimizeTours()));
+}
+
+process.on('unhandledRejection', err => {
+  console.error(err.message);
+  process.exitCode = 1;
+});
+main(...process.argv.slice(2));
```
Comment (subfuzion) [inline routeoptimization/snippets/optimizeTours.js:53]: Acknowledging that some inconsistency currently is present in existing samples, nevertheless please remove the process handler.

It doesn't add value to the sample; an unhandled rejection will terminate the process and set the exitCode and provide a full stack trace.

If it's important to handle, you can simultaneously also address the warning that the call to main on line 53 ignores the returned promise by using [Promise.prototype.catch()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch)). For example:

```javascript
main(...process.argv.slice(2)).catch(err => {
  console.error(err);
  process.exitCode = 1;
});
```

**Review by subfuzion — CHANGES_REQUESTED** [review]
Hi @bshi Thanks for the PR; please address the changes I've requested and I'll do another review pass afterwards.

File: CODEOWNERS (hunk 1 above)
Comment (bshi) [reply CODEOWNERS:72]: Confirmed, see https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9367#discussion_r1649561948

File: routeoptimization/snippets/optimizeTours.js (hunk 2 above)
Comment (bshi) [reply routeoptimization/snippets/optimizeTours.js:53]: It was cargo culted.  Removed.

File: CODEOWNERS (hunk 1 above)
Comment (subfuzion) [reply CODEOWNERS:72]: Thanks, I had just confirmed the team myself here: https://github.com/orgs/GoogleCloudPlatform/teams/geo-routeoptimization

Comment (iennae) [reply CODEOWNERS:72]: I've added the team to the repository with write access. This should allow you to work off a clone as well. Thanks!

**Review by subfuzion — APPROVED** [review]

**Pull request merged**